backend/tests/test_geoserver.py

```diff
@@ -0,0 +1,76 @@
+import os
+import pytest
+import asyncio
+from overstroomik_service.geoserver import Geoserver
+import warnings
+
+@pytest.mark.skipif(bool(os.getenv("GEOSERVER_HOST")) == False, reason="requires environment variable GEOSERVER_HOST (localhost)")
+async def test_get_data() -> None:    
+    
+    geoserver_host = os.getenv("GEOSERVER_HOST")
+    
+    geoserver = Geoserver()
+    status, data = await geoserver.get_data(
+        rd_x=158492.416,
+        rd_y=502424.151,
+        geoserver_url=f"http://{geoserver_host}:8080/geoserver")
+
+    assert status == "no-error"
+    assert data.flood_type == "binnen dijkring, mogelijk overstroombaar"
+    assert data.maximum_water_depth == 2.0899999141693115
+    assert data.probability_of_flooding is None
+    assert data.evacuation_percentage == 55
+    assert data.safety_board_id == 25
+        
+
+async def test_no_service() -> None:    
+    
+    geoserver = Geoserver()
+    status, data = await geoserver.get_data(
+        rd_x=158492.416,
+        rd_y=502424.151,
+        geoserver_url=f"http://unknown:8080/geoserver")
+
+    assert status == "Geen response geoserver"
+        
+
+@pytest.mark.skipif(bool(os.getenv("GEOSERVER_HOST"))==False, reason="requires environment variable GEOSERVER_HOST")
+def test_get_api_url_from_rd() -> None:
+    
+    geoserver_host = os.getenv("GEOSERVER_HOST")
+    
+    # test 1
+    geoserver = Geoserver()
+    data = geoserver.get_api_url_from_rd(
+        rd_x=158492.416,
+        rd_y=502424.151,
+        geoserver_url=f"http://{geoserver_host}:8080/geoserver")
+
+    assert f"http://{geoserver_host}:8080/geoserver" in data.get("api_url", "")
+    assert data.get("x", 0) == 157858.416
+    assert data.get("y", 0) == 134556.849
+    assert data.get("width", 0) == 283666.0
+    assert data.get("height", 0) == 330387.0
+    assert data.get("valid", False) is True
+
+    # test 2
```

Comment:
Document for these cases what/why you're actually testing. I also see now why there's a whole object returned from the `get_api_url_from_rd` function, it's the extra data for testing here.